Spanner/src/Middleware/RequestIdHeaderMiddleware.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/*
+ * Copyright 2025 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\Cloud\Spanner\Middleware;
+
+use Google\ApiCore\Call;
+use Google\ApiCore\Middleware\MiddlewareInterface;
+
+/**
+ * Middleware that adds the RequestId header to each rpc call made by spanner
+ */
+class RequestIdHeaderMiddleware implements MiddlewareInterface
+{
+    private const REQUEST_ID_HEADER_NAME = 'x-goog-spanner-request-id';
+    private const VERSION = 1;
+    private static string $process;
+    private static int $client = 0;
+    private int $channel;
+    private int $request = 1;
+
+    /** @var callable */
+    private $nextHandler;
+
+    public function __construct(callable $nextHandler, int $channelId)
+    {
+        $this->nextHandler = $nextHandler;
+        $this->channel = $channelId;
+        self::$client++;
```

Comment:
This will exhibit unwanted behavior. When it's incremented, because it's static, ALL clients will get incremented. E.g:

```php
$m1 = new RequestIdTokenMiddleware();
var_dump($m1); // has "$client" of "1"
$m2 = new RequestIdTokenMiddleware();
var_dump($m2); // has "$client" of "2"
var_dump($m1); // now ALSO has "$client" of "2", when we would expect it to still be "1"
```

The way to solve this would be to use a secondary instance variable instead:

```php
private static int $totalClients = 0;
private int $client;

public function __construct(callable $nextHandler, int $channelId)
{
    $this->client = self::$client++;
    // ...
}
```

Also, we should add a test for this!